addon/services/session.js

```diff
@@ -52,10 +52,14 @@ export default class Service extends SessionServiceESA {
     const isExpired = expireTime && expireTime <= new Date().getTime();
 
     if (this.isAuthenticated && isExpired) {
-      return yield this.session.authenticate("authenticator:oidc", {
-        redirectUri: this.redirectUri,
-        isRefresh: true,
-      });
+      try {
+        return yield this.session.authenticate("authenticator:oidc", {
+          redirectUri: this.redirectUri,
+          isRefresh: true,
+        });
+      } catch (e) {
+        console.warn("Token is invalid. Re-authentification is required.");
+      }
     }
```

Comment:
This `try ... catch` block catches the error thrown by the *oidc authenticator* and let's `ember-simple-auth` handle the invalid session afterwards. Without it, the `refreshAuthentication` task would simply fail and the `super.requireAuthentication` call could not trigger the `invalidateSession` event internally. 

With it, the invalidation is happening and finally the host app can decide what to do in the `handleUnauthorized` override! :tada:  

```js
 async requireAuthentication(transition, routeOrCallback) {
    await this.refreshAuthentication.perform();
    return super.requireAuthentication(transition, routeOrCallback);
  }
```

Reply:
Just for the record: [ember-concurrency tasks will throw on awaited promises that reject.](http://ember-concurrency.com/docs/task-function-syntax/)

> If you await a promise that rejects, the task function will throw the rejected value (likely an exception object) from the point in task function where the rejecting promise was awaited. This means you can use try {} catch(e) {} finally {} blocks, just as you would for code that runs synchronously.